plane/src/proxy/proxy_service.rs

```diff
@@ -181,6 +196,10 @@ impl RequestHandler {
             return Err(ProxyError::InvalidConnectionToken);
         };
 
+        if request_rewriter.get_subdomain(&route_info.cluster)? != route_info.subdomain.as_deref() {
+            return Err(ProxyError::InvalidSubdomain);
+        }
```

Comment:
let's add a `tracing::warn!` here for now, we can remove it later if it gets noisy